openjdk.test.math/src/test.math/net/adoptopenjdk/test/math/MathAPITestGenerator.java

```diff
@@ -239,32 +319,62 @@ private void generate()
 		{
-			if(Math.abs(doubleTestValues[i]) <= 1 && Math.abs(doubleTestValues[i]) >= 0)
+			if(StrictMath.abs(doubleTestValues[i]) <= 1 && StrictMath.abs(doubleTestValues[i]) >= 0)
 			{
-				System.out.println("assertEquals(\"acos(double)["+testCount+++"] ::\", " + Math.acos(doubleTestValues[i]) + ", Math.acos("+doubleTestStrings[i]+"));");
+				System.out.println("assertEquals(\"acos(double)["+testCount+++"] ::\", " + StrictMath.acos(doubleTestValues[i]) + ", StrictMath.acos("+doubleTestStrings[i]+"));");
+				if ( Double.isInfinite(StrictMath.acos(doubleTestValues[i])) ) {
+					System.out.println("assertEquals(\"acos(double)["+testCount+++"] ::\", " + StrictMath.acos(doubleTestValues[i]) + ", Math.acos("+doubleTestStrings[i]+"));");
+				}
+				else {
+					dTolerance = StrictMath.ulp(StrictMath.acos(doubleTestValues[i]));
+					System.out.println("assertTrue(\"acos(double)["+testCount+++"] ::\", " + (StrictMath.acos(doubleTestValues[i]) - dTolerance) + " <= Math.acos("+doubleTestStrings[i]+") && " + (StrictMath.acos(doubleTestValues[i]) + dTolerance) + " >= Math.acos("+doubleTestStrings[i]+"));");
+					System.out.println("assertTrue(\"acos(double)["+testCount+++"] ::\", " + (StrictMath.acos(doubleTestValues[i]) - StrictMath.ulp(StrictMath.acos(doubleTestValues[i]))) + " <= Math.acos("+doubleTestStrings[i]+") && " + (StrictMath.acos(doubleTestValues[i]) + StrictMath.ulp(StrictMath.acos(doubleTestValues[i]))) + " >= Math.acos("+doubleTestStrings[i]+"));");
+				}
+
 			}
 		}
-		for(int i = 0; i < floatTestValues.length; i++)
+		for (int i = 0; i < floatTestValues.length; i++)
 		{
-			if(Math.abs(floatTestValues[i]) <= 1 && Math.abs(floatTestValues[i]) >= 0)
+			if(StrictMath.abs(floatTestValues[i]) <= 1 && StrictMath.abs(floatTestValues[i]) >= 0)
 			{
-				System.out.println("assertEquals(\"acos(float)["+testCount+++"] ::\", " + Math.acos(floatTestValues[i]) + ", Math.acos("+floatTestStrings[i]+"));");
+				System.out.println("assertEquals(\"acos(float)["+testCount+++"] ::\", " + StrictMath.acos(floatTestValues[i]) + ", StrictMath.acos("+floatTestStrings[i]+"));");
+				if ( Double.isInfinite(StrictMath.acos(floatTestValues[i])) ) {
+					System.out.println("assertEquals(\"acos(float)["+testCount+++"] ::\", " + StrictMath.acos(floatTestValues[i]) + ", Math.acos("+floatTestStrings[i]+"));");
+				}
+				else {
+					dTolerance = StrictMath.ulp(StrictMath.acos(floatTestValues[i]));
+					System.out.println("assertTrue(\"acos(float)["+testCount+++"] ::\", " + (StrictMath.acos(floatTestValues[i]) - dTolerance) + " <= Math.acos("+floatTestStrings[i]+") && " + (StrictMath.acos(floatTestValues[i]) + dTolerance) + " >= Math.acos("+floatTestStrings[i]+"));");
+				}
 			}
 		}
-		for(int i = 0; i < longTestValues.length; i++)
+		for (int i = 0; i < longTestValues.length; i++)
 		{
-			if(Math.abs(longTestValues[i]) <= 1 && Math.abs(longTestValues[i]) >= 0)
+			if(StrictMath.abs(longTestValues[i]) <= 1 && StrictMath.abs(longTestValues[i]) >= 0)
```

Comment:
Naive question: The fact that we are changing this testcase from using Math to StrictMath APIs here, does this, in any way, mean that we are deciding to stop testing what this test was originally testing (e.g., java.lang.Math APIs), and introducing a new testcase (e.g. test java.lang.StrictMath APIs)?

Reply:
The test now tests both APIs.  This PR uses the StrictMath result from the reference jdk as the expected result rather than the Math result because the StrictMath result should be the same across all implementations but the Math result may not be.  The spec says the Math result (for most APIs) should be with 1 ulp of the StrictMath result and that is what the test now checks.  So we are still testing the Math APIs - but now against an accurate expected result.  I added the StrictMath tests because it was trivial to do so, and if two implementations gave different StrictMath results it would be good to know that.

Reply:
Regarding the line you draw attention to above (line 354), the check is there to avoid passing an invalid value to the subsequent operation.  abs() doesn't allow for a variation in precision, so it shouldn't matter whether it's the Math or StrictMath value which is checked.  I used StrictMath on the principle that the StrictMath operations of the reference jdk are those now being used as the basis for the expected results.

Reply:
Thanks for clarifying.